lighthouse-core/report/v2/renderer/report-features.js

```diff
@@ -0,0 +1,273 @@
+/**
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+/* globals Logger */
+/* eslint-env browser */
+
+class ReportFeatures {
+
+  /**
+   * @param {!Document} document
+   */
+  constructor(document) {
```

Comment:
this is a port from v1. Only difference is that the constructor does not automatically attache event listeners. You have to call `attach()` now.